pytorch_lightning/utilities/auto_restart.py

```diff
@@ -570,3 +571,12 @@ def reload_dataloader_state_dict(dataloader: DataLoader, state_dict: Dict[str, A
 
     else:
         raise MisconfigurationException("This shouldn't happen. Please, open an issue on PyTorch Lightning Github.")
+
+
+@runtime_checkable
+class _SupportsStateDict(Protocol):
+    def state_dict(self) -> Dict[str, Any]:
+        ...
```

Comment:
rather add a docstring and explain what it shall do...

Reply:
This is a protocol class, so better to use ... and it is for private usage.